[NUI] Add some accessibility features for new components

Seoyeon2Kim opened the pull request:
### Description of Change ###
- Introduce essential additions to support accessibility features for OneUI Components. 
  1) AccessibilityStateV2 
  2) AccessibilityRoleV2
  3) AccessibilityIsModal


User can use these features like this :
```
      SetAccessibilityRoleV2(AccessibilityRoleV2.Button);
      AccessibilityIsModal = true;

      var state = GetAccessibilityStatesV2();
      state[AccessibilityStateV2.Checked] = isChecked;
      SetAccessibilityStatesV2(state);
```


### API Changes ###
- N/A

hinohie commented on src/Tizen.NUI/src/public/BaseComponents/ViewAccessibility2.cs:
```diff
@@ -0,0 +1,289 @@
+            }
+        }
+
+        AccessibilityStates2(int states)
```
```suggestion
        [EditorBrowsable(EditorBrowsableState.Never)]
        public AccessibilityStates2(int states)
```

I don't know how we define to struct's constructor level but.. It looks browsable  never required here.

hinohie commented on src/Tizen.NUI/src/public/BaseComponents/ViewAccessibility2.cs:
```diff
@@ -0,0 +1,289 @@
+            _bitMask = (uint)states;
+        }
+
+        public static explicit operator int(AccessibilityStates2 state)
```
```suggestion
        [EditorBrowsable(EditorBrowsableState.Never)]
        public static explicit operator int(AccessibilityStates2 state)
```

hinohie commented on src/Tizen.NUI/src/public/BaseComponents/ViewAccessibility2.cs:
```diff
@@ -0,0 +1,289 @@
+            return (int)state._bitMask;
+        }
+
+        public static explicit operator AccessibilityStates2(int states)
```
```suggestion
        [EditorBrowsable(EditorBrowsableState.Never)]
        public static explicit operator AccessibilityStates2(int states)
```

hinohie commented on src/Tizen.NUI/src/public/BaseComponents/ViewAccessibility2.cs:
```diff
@@ -0,0 +1,289 @@
+    /// <summary>
+    /// AccessibilityState2 is an enumeration that represents the states of the view to send to accessibility service.
+    /// </summary>
+    public enum AccessibilityState2
```
```suggestion
    [EditorBrowsable(EditorBrowsableState.Never)]
    public enum AccessibilityState2
```

Also, We usually suggest to add never browsable for all each enums.

Seoyeon2Kim replied:
One more question!
I think we should add each value of this enum, right?

Seoyeon2Kim commented:
Thank you for checking to add `[EditorBrowsable(EditorBrowsableState.Never)]` tags, @hinohie 
I don't decide the naming of these features, so I couldn't check the tag all.
But, I hope `ACR Required` tag doesn't be added anymore since the new forced-push commit !

Seoyeon2Kim commented on src/Tizen.NUI/src/public/BaseComponents/ViewAccessibility2.cs:
```diff
@@ -0,0 +1,289 @@
+    public struct AccessibilityStates2
+    {
+        const uint _on = 1;
+        uint _bitMask;
```
Oh missed it!
I'll add it in the next commit ! (Sorry~)

Seoyeon2Kim commented on src/Tizen.NUI/src/public/BaseComponents/ViewAccessibility2.cs:
```diff
@@ -0,0 +1,289 @@
+        /// <returns>The states<see cref="AccessibilityStates2"/> of the view</returns>
+        public AccessibilityStates2 GetAccessibilityStates2()
+        {
+
```
nice catch

rabbitfor reviewed (commented):
Thank you for your work :)
One minor request. Could you seperate files by classes? One class, one file would be good.

bshsqa commented on src/Tizen.NUI/src/public/BaseComponents/ViewAccessibilityV2.cs:
```diff
@@ -0,0 +1,328 @@
+    public struct AccessibilityStatesV2
+    {
+        const uint _on = 1;
+        private uint _bitMask;
```
Question.
Which convention rule is right for field?
using "_" or lower charactor?

Seoyeon2Kim replied:
I just follow the Dotnet coding style below :
https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#camel-case

rabbitfor replied:
https://confluence.sec.samsung.net/display/GFX/NUI+Ground+Rule
It is defined NUI coding rule. (Alhtough most NUI code are not following this)

> C# coding style follows dotnet runtime guidelines
https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md
In addition to rule3, use "c_" prefix for const (or readonly) field
c_somePrivateConstFieldName

Seoyeon2Kim replied:
I think I also need to add `c_` prefix for `_on` variable, don't I?

bshsqa commented on src/Tizen.NUI/src/public/BaseComponents/ViewAccessibilityV2.cs:
```diff
@@ -0,0 +1,328 @@
+        /// Gets the state at the specsified name
+        /// </summary>
+        /// <param name="state">The name of the state</param>
+        /// <returns>The state af the specified name</returns>
```
Return type is boolean. It means state? or whether the state is included or not?

Seoyeon2Kim replied:
Done.

bshsqa commented on src/Tizen.NUI/src/public/BaseComponents/ViewAccessibilityV2.cs:
```diff
@@ -0,0 +1,328 @@
+        private uint _bitMask;
+
+        /// <summary>
+        /// Gets the state at the specsified name
```
This property has set too. Isn't it?

Seoyeon2Kim replied:
Done.

Seoyeon2Kim commented:
> Thank you for your work :) One minor request. Could you seperate files by classes? One class, one file would be good.

Thanks for your review.
In the file, there's only one class, so if you mean it's each feature, including enumerations and structure, can I separate them, such as `ViewAccessibilityRoleV2`, `ViewAccessibilityStateV2`, and so on?

rabbitfor commented:
> > Thank you for your work :) One minor request. Could you seperate files by classes? One class, one file would be good.
> 
> @rabbitfor Thanks for your review. In the file, there's only one class, so if you mean it's each feature, including enumerations and structure, can I separate them, such as `ViewAccessibilityRoleV2`, `ViewAccessibilityStateV2`, and so on?

Yes please separate all features including enum, struct :)

Seoyeon2Kim commented:
There are lots of commits in this one PR, so we should merge this PR with `Squash and merge` button :)

Seoyeon2Kim commented:
I've never added this API :(
Why is the ACR required tag added here?

> Public API Changed
> 
> Please follow the ACR process for the changed API below.
> Added: 1, Removed: 0, Changed: 0
> Added
> 
> + /// <since_tizen>none</since_tizen
> + System.Void Tizen.Multimedia.Recorder::Finalize()
>

dongsug-song commented:
> I've never added this API :( Why is the ACR required tag added here?
> 
> > Public API Changed
> > Please follow the ACR process for the changed API below.
> > Added: 1, Removed: 0, Changed: 0
> > Added
> > 
> > * /// <since_tizen>none</since_tizen
> > * System.Void Tizen.Multimedia.Recorder::Finalize()

I think something wrong. I will remove the tag manually.
thank you.
